coderd/userauth.go

```diff
@@ -289,8 +289,17 @@ func (api *API) userOIDC(rw http.ResponseWriter, r *http.Request) {
 		}
 		username = httpapi.UsernameFrom(username)
 	}
+	// Check if one or comma delimited list of allowed domains is provided.
+	// If a suffix matches, break and continue, otherwise error.
 	if api.OIDCConfig.EmailDomain != "" {
-		if !strings.HasSuffix(strings.ToLower(email), strings.ToLower(api.OIDCConfig.EmailDomain)) {
+		ok = false
+		for _, domain := range strings.Split(api.OIDCConfig.EmailDomain, ",") {
```

Comment:
Perhaps we should turn the [DeploymentConfig EmailDomain](https://github.com/coder/coder/blob/883cf8afa94903b0666dae31bc21cd935a6a30c2/cli/deployment/config.go#L218-L222) and `OIDCConfig.EmailDomain` into `[]string` instead? Otherwise a user wouldn't know that multiple values are possible.

Reply:
I have updated to EmailDomains and []string